core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1320,6 +1325,9 @@ default Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
      */
     default boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, Type versioning)
     {
+        if (getTableHandle(session, tableName) == null) {
+            return true;
```

Comment:
why true?

Reply:
That's an unfortunate consequence of the fact that `isSupportedVersionType`is checked before table existence.

But, a non-existent table supports all versioning schemes, right?

Reply:
> That's an unfortunate consequence of the fact that `isSupportedVersionType` is checked before table existence.

The interface method shouldn't make suppositions  related to the order in which is employed in the client code.

> But, a non-existent table supports all versioning schemes, right?

I tend to say no.
In such a situation I'd rather advocate for throwing an exception stating that the table does not exist.

Reply:
> The interface method shouldn't make suppositions related to the order in which is employed in the client code.

i generally agree, unless there is some design driving force that requires otherwise.
(eg avoiding getting table handle twice)

> In such a situation I'd rather advocate for throwing an exception stating that the table does not exist.

That's exactly what am i doing here.

Reply:
> (eg avoiding getting table handle twice)

Actually here you are adding an extra call to `getTableHandle` as part of `isSupportedVersionType` (but just for one connector).

What if we throw "table-not-found" exception directly from `isValidTableVersion`? I think this is what @findinpath was suggesting.

Reply:
Looks good now.